### PR TITLE
Loadtest async and erc20

### PR DESCRIPTION
**Pull request by BaldyAsh**

### CODE REVIEW

File: contracts/scripts/testnet-deploy.ts [hunk 1]
```diff
@@ -72,7 +73,22 @@ async function main() {
         franklinAddress = franklin.address;
 
         await governance.setValidator(process.env.OPERATOR_ETH_ADDRESS, true);
-        await addTestERC20Token(wallet, governance);
+        const erc20 = await addTestERC20Token(wallet, governance);
+
+        // Setup test wallets
+        if (process.env.ETH_NETWORK === "localhost") {
+            let i = 0;
+            while (i < +process.env.TEST_WALLETS_TOTAL) {
```
Comment (dvush) [inline contracts/scripts/testnet-deploy.ts]: there is for loop in js

```javascript
for(let i = 0; i < 10; i++) {
/// ...
}
```

File: contracts/scripts/testnet-deploy.ts [hunk 2]
```diff
@@ -72,7 +73,22 @@ async function main() {
         franklinAddress = franklin.address;
 
         await governance.setValidator(process.env.OPERATOR_ETH_ADDRESS, true);
-        await addTestERC20Token(wallet, governance);
+        const erc20 = await addTestERC20Token(wallet, governance);
+
+        // Setup test wallets
+        if (process.env.ETH_NETWORK === "localhost") {
```
Comment (dvush) [inline contracts/scripts/testnet-deploy.ts]: Bad sign -- difference between localhost and testnet.

File: bin/loadtest.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+. .setup_env
+
+for ((i=0;i<$TEST_WALLETS_TOTAL;i++)); do
+    cd js/tests && WALLET=$i yarn loadtest &
```
Comment (dvush) [inline bin/loadtest.sh]: Bad sign -- complex setup of multiple wallets required for this script, but it is just detail of the implementation.

File: etc/env/dev.env.example [hunk 4]
```diff
@@ -1,12 +1,8 @@
-# Loadtest
-
-LOADTEST_N_CLIENTS=10
-LOADTEST_TPS=100
-
 # Mnemonic can be generated here: https://bitcoinqrcodegenerator.win/bip39/
 
 MNEMONIC="fine music test violin matrix prize squirrel panther purchase material script deal"
 
+TEST_WALLETS_TOTAL=25
```
Comment (dvush) [inline etc/env/dev.env.example]: detail of the implementation

File: js/tests/loadtest.ts [hunk 5]
```diff
@@ -44,173 +46,136 @@ let TRANSFERS_TOTAL = 100;
...
-            i++;
+        promises = [];
+        results = [];
+        failedPromises = [];
+
+        for (const wallet of syncWallets) {
+            const i = syncWallets.indexOf(wallet);
+            promises.push(withdraw(wallet, ethWallets[i], ["ETH", ERC20_TOKEN]));
+        }
+        await Promise.all(promises.map(reflect));
```
Comment (dvush) [inline js/tests/loadtest.ts]: `Promise.all` will reject if one of them will reject -- it is not acceptable. Use something like `Promise.allSettled()`

File: js/tests/loadtest.ts [hunk 6]
```diff
@@ -120,6 +121,14 @@ const FEE_DIVISOR = 50;
 
 })();
 
+async function createRandomZKSyncWallet(ethWallet: ethers.Wallet, provider: Provider, ethProxy: ETHProxy) {
```
Comment (dvush) [inline js/tests/loadtest.ts]: why did you add this? this should be deterministic. please revert back to `fromEthSigner` where signers are derived from mnemonic.

Comment (BaldyAsh) [reply js/tests/loadtest.ts]: `fromEthSigner` func from zksync.js uses a static seed to generate a private key. And in order to be able to run the loadtest in several terminal tabs at the same time, I need different accounts in order to avoid a nonce collision. Therefore, I wrote a random seed generation in this function
